persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -315,7 +317,7 @@ upsertSql' ent updateVal = T.concat
                            , " RETURNING ??"
                            ]
     where
-      wher = T.intercalate " AND " $ map singleCondition $ entityUniques ent
+      wher = T.intercalate " AND " $ map singleCondition $ NEL.toList uniqs
```

Comment:
The issue that prompted this change is documented in #856 . Basically, the type `UpsertBy` in the tests has two unique defs. The old implementation summoned up all the uniquedefs, and this was OK since the `upsert` function would blow up at runtime before generating this unless the entity had exactly *one* unique key.

This change allows the user to provide the unique keys to upsert by. This enables backend-specific upsert-by which should be faster and safer.